### PR TITLE
Adding icons for infill stations

### PR DESCRIPTION
**Pull request by vickymmcd**
Screenshot from 2022-10-24 21-48-52

### CODE REVIEW

File: src/components/JourneyTimeline/JourneyTimeline.tsx [hunk 1]
```diff
@@ -24,6 +24,51 @@ const expandControlSpacingPx = 100;
...
+        "place-rr-cohasset-center",
+        "place-rr-braintree-highlands",
+        "place-rr-bridgewater-center",
+        "place-rr-mboro-centre-st",
+        "place-rr-weymouth-col-sq",
+        "place-rockland-n-abington",
+        "place-kingston-jct",
+        "place-plymouth-center",
+    ]);
+    const infillInRoute = {
```
Comment (vickymmcd) [inline src/components/JourneyTimeline/JourneyTimeline.tsx:58]: @idreyn do we have a list of these somewhere? I know the ones for the northern lines, but I'm not as familiar with the southern lines

Comment (idreyn) [reply src/components/JourneyTimeline/JourneyTimeline.tsx:58]: I'm not sure if it's comprehensive but I tried to put them in italics in the spec doc!

**Review by idreyn — CHANGES_REQUESTED** [review]
Thanks for taking this on, and dealing with the awful headache that is `.module.scss` files. There are a few opportunities to de-duplicate the added code that I outlined inline, but I would suggest doing the highest-leverage thing and just creating a new `StationName.tsx` component like this:

```tsx
type Props = {
    showInfillStatus?: boolean; // only show the (+) if this is true
    asLink?: boolean; // only render as a link if this is true
}

const StationName = (props: Props) => {
    const { showInfillStatus = true, asLink = false } = props;
    // render an <a href> if asLink, else a <span> ...
}
```

This would live alongside a `StationName.module.scss` file that contains the CSS you added for the plus icon, tooltip, etc.

Then we can just toss that in wherever we render a station name — so far it's just in `JourneyTimeline` and `StationListing`, and _possibly_ in the `JourneyPicker` dropdowns.

(You can skip the `asLink` part for now since we don't actually have a station page yet, but someone is working on that so we'll need it eventually)

I know this is a lot to drop on you in code review, but how does that sound? I think the result will be really clean and easy+fun to reuse across the codebase!

File: src/components/JourneyTimeline/JourneyTimeline.tsx (hunk 1 above)
Comment (idreyn) [reply src/components/JourneyTimeline/JourneyTimeline.tsx:58]: Aside from the ones that are existing rapid transit stations, I believe they're all in `infill_stations.py` in the schedule generator. Also, I meant to make them all start with `place-rr-` and I must have slipped up on the last few. We could easily fix that and then use the convention of `place-rr-` (or `place-infill-`) implying an infill station.

**Comment by vickymmcd** [comment]
> 
@idreyn This is great, thank you so much for the feedback (and sorry for my delay in addressing it)! I have just started working on the changes and I'm wondering about using the `StationName` component in `StationListing`, specifically in regards to your suggestion around making the station name link-able. Isn't the name shown in `StationListing` already a link used to select that station as part of your journey? Or perhaps I am misunderstanding where `StationListing` is used.

**Review by idreyn — APPROVED** [review]

**Pull request merged**